### PR TITLE
FQCN roles

### PR DESCRIPTION
**Pull request by thekad**
Roles referencing other collection roles should also FQCN their include/import

Test-Hints: assisted

### CODE REVIEW

File: roles/boot_disk/tasks/main.yml [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-  set_fact:
+  ansible.builtin.set_fact:
     hosts: "{{ groups['masters'] + groups['workers'] | default([]) }}"
   when: hosts is not defined
 
 - name: Boot from disk for "{{ hostvars[item]['vendor'] }}"
-  include_role: 
-    name: "vendors/{{ hostvars[item]['vendor'] | lower }}"
+  ansible.builtin.include_role:
+    name: "redhatci.ocp.vendors/{{ hostvars[item]['vendor'] | lower }}"
```
Comment (tonyskapunk) [inline roles/boot_disk/tasks/main.yml:11]: I'm not sure this is correct, I think it should use FQCN:

```suggestion
    name: "redhatci.ocp.vendors.{{ hostvars[item]['vendor'] | lower }}"
```

File: roles/hco_setup/README.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ As a task:
 ```yaml
 - name: "Deploy hco-operator and test it"
   include_role:
```
Comment (tonyskapunk) [inline roles/hco_setup/README.md:40]: Let's use `ansible.builtin.include_role` as in the other READMEs

File: roles/install_cluster/tasks/main.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 - name: Patch discovered hosts
   include_role:
```
Comment (tonyskapunk) [inline roles/install_cluster/tasks/main.yml:25]: Should we  use `ansible.builtin.include_role` here?

File: roles/merge_registry_creds/README.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ The `mrc_auth_data` variable contains the result of combining all the authentica
 ```yaml
 - name: "Combine registry auth secrets"
   include_role:
```
Comment (tonyskapunk) [inline roles/merge_registry_creds/README.md:21]: `ansible.builtin.include_role` ?

**Review by tonyskapunk — CHANGES_REQUESTED** [review]

File: roles/boot_disk/tasks/main.yml (hunk 1 above)
Comment (thekad) [reply roles/boot_disk/tasks/main.yml:11]: I am not sure that's correct either, but your suggestion seems better

**Review by thekad — COMMENTED** [review]
Most of the errors came from a stashed chunk I applied before my other PR got merged, so I had a regression on some of my FQCN includes before. I have updated the READMEs with your suggestions as well

File: roles/validate_inventory/tasks/vendor.yml [hunk 5]
```diff
@@ -9,8 +9,8 @@
   loop: "{{ groups['nodes'] }}"
 
 - name: "Check vendors role exists"
-  include_role: 
-    name: "vendors/{{ hostvars[item]['vendor'] | lower }}"
+  include_role:
```
Comment (tonyskapunk) [inline roles/validate_inventory/tasks/vendor.yml:12]: Could you also use FQCN in the include_role? also the following files don't have FQCN in the include_role let's change them too

roles/generate_discovery_iso/tasks/static.yml
roles/add_day2_node/tasks/main.yml

Thanks

Comment (thekad) [reply roles/validate_inventory/tasks/vendor.yml]: all roles coming from crucible need to be audited for FQCN, but I'm not doing that just now. I have changed the files you mentioned, but let's hold off on the rest for another PR.

Comment (tonyskapunk) [reply roles/validate_inventory/tasks/vendor.yml]: Thanks! I was referring only to the include_role part :smile: but I see you have changed them all, sorry if I was not clear.

**Review by tonyskapunk — APPROVED** [review]
LGTM

I see now the sub-roles are listed with FQCN https://www.distributed-ci.io/jobs/3bc93b41-0acf-4223-ab03-2f42990c48b3/jobStates?sort=date&task=19132701-0fbb-4f7a-a102-0982bf14e741

File: roles/boot_iso/tasks/main.yml [hunk 6]
```diff
@@ -8,8 +8,8 @@
   delegate_to: bastion
 
 - name: Boot ISO for "{{ hostvars[inventory_hostname]['vendor'] }}"
-  include_role: 
-    name: "vendors/{{ hostvars[inventory_hostname]['vendor'] | lower }}"
+  ansible.builtin.include_role:
+    name: "redhatci.ocp.vendors.{{ hostvars[inventory_hostname]['vendor'] | lower }}"
```
Comment (nocturnalastro) [inline roles/boot_iso/tasks/main.yml:12]: I will have to check this still works.

File: roles/install_cluster/tasks/main.yml [hunk 7]
```diff
@@ -210,4 +210,4 @@
   until: cluster.json.status in cluster_installation_states
   retries: 5
   delay: 60
-  
+...
```
Comment (nocturnalastro) [inline roles/install_cluster/tasks/main.yml:213]: Note sure if there should be triple dots here. Is this a yaml thing?

Comment (thekad) [reply roles/install_cluster/tasks/main.yml:213]: yeah this is just yaml syntax for "end of document"

File: roles/boot_iso/tasks/main.yml (hunk 6 above)
Comment (nocturnalastro) [reply roles/boot_iso/tasks/main.yml:12]: Worked fine for me using a vendor kvm.

**Review by nocturnalastro — APPROVED** [review]

**Review by nocturnalastro — APPROVED** [review]

**Review by nocturnalastro — APPROVED** [review]

**Pull request merged**